nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala

```diff
@@ -154,9 +154,20 @@ abstract class NirCodeGen
       else Some(genTypeName(sym.superClass))
 
     def genClassAttrs(sym: Symbol): Attrs = {
-      def pinned =
-        if (!isModule(sym) || isExternModule(sym)) Seq()
-        else Seq(Attr.PinAlways(genMethodName(sym.asClass.primaryConstructor)))
+      def pinned = {
+        val ctor =
+          if (!isModule(sym) || isExternModule(sym)) Seq()
+          else
+            Seq(Attr.PinAlways(genMethodName(sym.asClass.primaryConstructor)))
```

Comment:
Constructors are pinned for module classes only. This is necessary due toe the fact that one never calls it explicitly, it's only being used inside the implementation of `module $name` opcode (i.e. in generated accessor that is injected by `ModuleLowering`). This way if you depend on module, you'll also depend on the module's constructor automatically.